crates/sui-source-validation-service/src/lib.rs

```diff
@@ -541,3 +513,48 @@ async fn api_route(
         )
     }
 }
+
+async fn check_version_header<B>(
+    headers: HeaderMap,
+    req: hyper::Request<B>,
+    next: Next<B>,
+) -> Response {
+    let version = headers
+        .get(SUI_SOURCE_VALIDATION_VERSION_HEADER)
+        .as_ref()
+        .and_then(|h| h.to_str().ok())
+        .map(|s| s.to_string());
+
+    let mut headers = HeaderMap::new();
+    headers.insert(
+        SUI_SOURCE_VALIDATION_VERSION_HEADER,
+        SUI_SOURCE_VALIDATION_VERSION.parse().unwrap(),
+    );
+
+    match version {
+        Some(v) if v != SUI_SOURCE_VALIDATION_VERSION => {
+            let error = format!(
+                "Unsupported version '{v}' specified in header \
+		 {SUI_SOURCE_VALIDATION_VERSION_HEADER}"
+            );
+            return (
+                StatusCode::BAD_REQUEST,
+                headers,
+                Json(ErrorResponse { error }).into_response(),
+            )
+                .into_response();
+        }
+        None => {
+            info!("No version set, using {SUI_SOURCE_VALIDATION_VERSION}");
```

Comment:
This may be a redundant/noisy log message, in practice, because each binary of your service only supports one version, so you would know by the process the logs are coming from which version it supports.

Reply:
agree, fixed up to not print this